monai/data/dataset.py

```diff
@@ -137,7 +137,7 @@ def __init__(
         self,
         data: Sequence,
         transform: Union[Sequence[Callable], Callable],
-        cache_dir: Optional[Union[Path, str]] = None,
+        cache_dir: Optional[Union[Path, str]],
```

Comment:
You want the user to explicitly say that they don't want caching? Makes sense to me given that this is `PersistentDataset`, but the change is backwards incompatible.

Reply:
existing code relying on the default  cache_dir=None is probably wrong anyway...